src/providers/dialogs-controller.ts

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { IDialogController } from '@uireact/dialog';
+
+export const useDialogController = (): IDialogController => {
+  const [openedDialogs, setOpenedDialogs] = React.useState<string[]>([]);
+
+  const openDialog = React.useCallback(
+    (dialogId: string) => {
+      const found = openedDialogs.filter((id) => id === dialogId);
+
+      if (!found || found.length === 0) {
+        const newOpenedDialogs = [...openedDialogs, dialogId];
+        setOpenedDialogs(newOpenedDialogs);
+      }
+    },
+    [openedDialogs, setOpenedDialogs]
+  );
+
+  const closeDialog = React.useCallback(
+    (dialogId: string) => {
+      const newOpenedDialogs = openedDialogs.filter((id) => id !== dialogId);
+      setOpenedDialogs(newOpenedDialogs);
+    },
+    [openedDialogs, setOpenedDialogs]
+  );
+
+  return {
+    openDialog,
+    closeDialog,
+    openedDialogs,
+  };
+};
```

Comment:
This is the controller that keeps the dialogs state inside the docs page, each client will have to create something similar to this and can do whatever they want to do for handling the dialogs state, in this case I'm using a useState to keep track of the opened dialogs but in a different client they could use the URL for instance or whatever other thing that fit their requirements.